packages/@o3r/extractors/builders/aggregate-migration-scripts/index.ts

```diff
@@ -0,0 +1,169 @@
+import { AggregateMigrationScriptsSchema } from './schema';
+
+/** Maximum number of steps */
+const STEP_NUMBER = 3;
+
+/** The content of the migration file plus the filename */
+interface MigrationFileEntry<T = any> {
+  fileName: string;
+  migrationObject: MigrationFile<T>;
+}
+
+/** A map that defines for each library the history of versions defined in all the original migration scripts */
+interface LibraryVersions {
+  [libName: string]: {libVersion: string; appVersion: string}[];
+}
+
+/**
+ * Get the content of all migration files matching a glob
+ * @param glob
+ */
+const getMigrationFiles = async (glob: string | string[]): Promise<MigrationFileEntry[]> => {
+  return await Promise.all((await globby(glob, {fs}))
+    .map(async (fileName) => ({
+      fileName,
+      migrationObject: JSON.parse(await readFile(fileName, {encoding: 'utf8'}))
+    }))
+  );
+};
+
+/**
+ * Check if the lib version is valid
+ * @param libVersion
+ */
+const isValidLibVersion = (libVersion: Partial<LibraryVersions[string][number]>): libVersion is LibraryVersions[string][number] =>
+  !!libVersion.libVersion && !!libVersion.appVersion;
+
+/**
+ * Get the history of versions for each library in chronological order
+ * @param migrationFiles
+ */
+const getLibrariesVersions = (migrationFiles: MigrationFileEntry[]): LibraryVersions => {
+  const libraries: LibraryVersions = {};
+  for (const file of migrationFiles) {
+    if (file.migrationObject.libraries) {
+      for (const [lib, version] of Object.entries<string>(file.migrationObject.libraries)) {
+        if (!libraries[lib]) {
+          libraries[lib] = [];
+        }
+        const libVersion = {
+          libVersion: semver.coerce(version)?.raw,
+          appVersion: semver.coerce(file.migrationObject.version)?.raw
+        };
+
+        if (isValidLibVersion(libVersion)) {
+          libraries[lib].push(libVersion);
+        }
+      }
+    }
+  }
+  // Sort and dedupe versions
+  Object.entries(libraries).forEach(([libName, libVersions]) => {
+    libVersions.sort((a, b) => semver.compare(a.appVersion, b.appVersion));
+    libraries[libName] = libVersions.filter((libVersion, index) =>
+      index < 1 || libVersions[index - 1].libVersion !== libVersion.libVersion);
+  });
+  return libraries;
+};
+
+/**
+ * Write the migration files on disk
+ * @param migrationFiles
+ * @param destinationPath
+ */
+const writeMigrationFiles = async (migrationFiles: MigrationFileEntry[], destinationPath: string) => {
+  const outputDirectory = resolve(destinationPath);
+  if (fs.existsSync(outputDirectory)) {
+    await rm(outputDirectory, {recursive: true});
+  }
+
+  for (const file of migrationFiles) {
+    const distFilePath = join(outputDirectory, basename(file.fileName));
+    if (!fs.existsSync(dirname(distFilePath))) {
+      await mkdir(dirname(distFilePath), {recursive: true});
+    }
+    await writeFile(distFilePath, JSON.stringify(file.migrationObject, null, 2) + '\n');
```

Comment:
Shouldn't we use EOL ?

Reply:
Not sure why this comment was flagged as `resolved` when the change has not been applied.
I would prefer `\n` instead of `EOL` because we force `LF` even on Windows

Reply:
I tested with `EOL` and faced the same conclusion so I marked that comment as resolved without answering sorry